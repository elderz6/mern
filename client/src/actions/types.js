export const GET_ITEMS = 'GET_ITEMS';
export const ADD_ITEM = 'ADD_ITEM';
export const DELETE_ITEM = 'DELETE_ITEM';
export const ITEMS_LOADING = 'ITEMS_LOADING';
export const GET_ITEMS_FAIL = 'GET_ITEMS_FAIL';

export const LOGIN_START = 'LOGIN_START';
export const LOGIN_LOAD = 'LOGIN_LOAD';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_FAIL = 'LOGIN_FAIL';

export const REGISTER_START = 'REGISTER_START';
export const REGISTER_LOAD = 'REGISTER_LOAD';
export const REGISTER_SUCCESS = 'REGISTER_SUCCESS';
export const REGISTER_FAIL = 'REGISTER_FAIL';
