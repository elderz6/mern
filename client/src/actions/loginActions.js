import axios from 'axios';
import {LOGIN_FAIL, LOGIN_LOAD, LOGIN_START, LOGIN_SUCCESS} from './types';

export const logUserStart = () => {
  return {
    type: LOGIN_START
  }
};

export const logUserSuccess = () => {
  return{
    type:LOGIN_SUCCESS
  }
};

export const logUserFail = error => {
  return{
    type:LOGIN_FAIL,
    error
  }
};

export const logUser = () => dispatch =>
{
  axios.post('/api/users');
};
