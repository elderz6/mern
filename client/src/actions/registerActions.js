import axios from 'axios';
import { REGISTER_START, REGISTER_FAIL, REGISTER_SUCCESS} from './types';

export const registerStart = () => {
  return {
    type: REGISTER_START
  }
};

export const registerSuccess = () => {
  return{
    type:REGISTER_SUCCESS
  }
};

export const registerFail = errors => {
  return{
    type:REGISTER_FAIL,
    errors
  }
};

export const registerUser = user => dispatch =>
{
  axios.post('./api/users', user)
  .then(res =>
    dispatch({
      type:REGISTER_SUCCESS,
      payload:res.data
  }))
  .catch(() =>
  {
    console.log('erro');
  });
};
