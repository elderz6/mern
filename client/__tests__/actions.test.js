import mockAxios from 'axios';
import thunk from 'redux-thunk'
import * as types from '../src/actions/types';
import * as actions from '../src/actions/itemActions';
import configureMockStore from 'redux-mock-store'
import fetchMock from 'fetch-mock'
const middlewares = [ thunk ];
const mockStore = configureMockStore(middlewares);
describe('Item Actions', ()=>
{
  it('should start to get items', ()=>
  {
    const store = mockStore({});
    store.dispatch(actions.getItems())
    const act = store.getActions();
    console.log(act);
    expect(act).toMatchSnapshot();
  });

  it('should work async', ()=>
  {
    const store = mockStore({})
    store.getState()
    const expected = {type:types.GET_ITEMS}
    store.dispatch(expected, actions.getItems())
    const act = store.getActions();
    console.log(act);
    expect(act).toEqual([expected])
    expect(act).toMatchSnapshot();
  })
});
