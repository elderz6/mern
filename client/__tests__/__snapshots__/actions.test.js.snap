// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Actions should start to get items 1`] = `Array []`;

exports[`Item Actions should work async 1`] = `
Array [
  Object {
    "type": "GET_ITEMS",
  },
]
`;
